src/commands/install.cr

```diff
@@ -1,31 +1,80 @@
 require "../spec"
+require "../lock"
 require "../manager"
 require "./command"
 
 module Shards
   module Commands
+    # OPTIMIZE: avoid updating GIT caches until required
     class Install < Command
-      getter :spec, :manager
+      getter :path, :manager
 
-      def initialize(path, groups)
-        @spec = Spec.from_file(path)
+      def initialize(@path, groups)
+        spec = Spec.from_file(path)
         @manager = Shards::Manager.new(spec, groups)
+        @locks = Lock.from_file(lock_file_path) if lock_file?
       end
 
-      # TODO: load dependencies from shards.lock if present,
-      #       otherwise resolve dependencies and save shards.lock
       def run
         manager.resolve
 
-        manager.packages.each do |package|
-          if package.installed?(loose: true)
-            Shards.logger.info "Using #{package.name} (#{package.version})"
+        if locks = @locks
+          install(manager.packages, locks)
+        else
+          install(manager.packages)
+        end
+
+        unless lock_file?
+          File.open(lock_file_path, "w") { |file| manager.to_lock(file) }
+        end
```

Comment:
The install command must regenerate the lock file in case any new package was installed.